doc/dev/crimson/poseidonstore.rst

```diff
@@ -0,0 +1,401 @@
+                                                                          |                        |
+                                                                          +------------------------+
+
+
+WAL
+---
+Each AG has a WAL.
+The datas written to the WAL are all metadta updates, free space update and data.
+Note that only data smaller than the predefined threshold needs to be written to the WAL.
+The larger data is written to the unallocated free space and its onode's extent_map is updated accordingly 
+(also on-disk extent map). We statically allocate WAL partition aside from data partition pre-configured.
+
+
+Partition
+---------
+Initially, PoseidonStore employs static allocation of partition. The number of sharded partitions
+is fixed and the size of each partition also should be configured before running cluster.
+
+
+Cache
+-----
+There are mainly two cache data structures; onode cache and block cache.
+Since PoseidonStore does not have the on-disk onode strucutre, there needs to be in-memory onode structure 
+for fast tracking of the objects.
+It looks like below.
+
+Onode {
+  metadata
+  xattr map <key, value>
+  omap <key, value>
+  extent_map
+}
+
+To fill the in-memory onode data structure, object meta, xattrs, and omap of the target object can be retrieved 
+using the respective PSD_OIDs.
+Block cache is used for caching a block contents and supporting transactions.
+For a transaction, all the updates to blocks (including object meta block, allocation bitmap block, data block) 
+are first performed in the in-memory block cache.
+After writing a transaction to the WAL, the dirty blocks are flushed to their respective locations in the 
+respective partitions.
+PoseidonStore can configure cache size for each type. Simple LRU cache eviction strategy can be used for both.
+
+
+Allocation Group (with cross-AG transaction)
+--------------------------------------------
+Entire disk space is divided into a number of equally sized chunks called Allocation Groups (AG).
+Each reactor thread in crimson-osd is responsible for a set of AGs (# of AGs / # of threads).
+Each AG has its own data structures to manage the disk partition.
+A collection is stored into an AG, not across AGs.
+The prefixes of the parent collection ID (original collection ID before collection splitting. That is, hobject.hash) 
+is hashed to map any collections to AGs allocated for the reactor thread.
+We can use BlueStore's approach for collection splitting, changing the number of significant bits for the collection prefixes.
+Because the prefixes of the parent collection ID do not change even after collection splitting, the mapping between 
+the collection and AG is maintained.
+The number of AGs may be configured to match the number of CPUs allocated for each disk so that each AG can hold 
+a number of objects large enough for cross-AG transaction not to occur.
```

Comment:
I need more detail on this part, I think you are missing a lot here that will matter hugely -- no cross ag transactions means every PG is local to a single AG and you cannot move less than a full PG between AGs.